scripts/utils/shell/shell.sh

```diff
@@ -0,0 +1,140 @@
+# shellcheck shell=bash
+
+# Title:          shell
+# Description:    A library for shell.
+# License-SPDX:   MIT
+# Function-Depends: logger
+
+
+
+##
+# shell__is_valid_shell_variable_name `variable_name`
+##
+shell__is_valid_shell_variable_name() {
+
+	# Case statements are slow for long strings
+	# 1000 is arbitrarily selected and needs testing
+	if [ "${#1}" -lt 1000 ]; then
+		# Use `${parameter:offset:length}`
+		local variable_name_rest="${1:1}" # 1:end
+		local variable_name_first_char="${1:0:1}" # 0:1
+
+		case "$variable_name_first_char" in
+			[a-zA-Z_])
+				case "$variable_name_rest" in
+					*[!a-zA-Z0-9_]*) return 1;;
+					*) return 0;;
+				esac;;
+			*) return 1;;
+		esac
+	else
+		# Do not use `grep` since `grep -z` does not work with
+		# start `^` and end `$` anchors
+		# We manually add everything to pattern space and print only
+		# if it matches the regex
+		[ -n "$(data__printf_string "${1}x" | sed -r -n -e '$!{:a;N;$!ba;}; /^[a-zA-Z_][a-zA-Z0-9_]+$/p')" ]
```

Comment:
In the case if this function is intensively used probably you want to use only bash builtins here to avoid performance hit.

Reply:
Read the `if` condition above, its only used if a variable name length exceeds `1000` characters, that normally does not happen, other than some of my crazy projects stuff where in rare conditions it can exceed.